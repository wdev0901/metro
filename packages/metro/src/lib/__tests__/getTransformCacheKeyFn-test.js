/**
 * Copyright (c) 2015-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @emails oncall+javascript_foundation
 * @format
 * @flow
 */

'use strict';

jest.mock('../../../package.json', () => ({
  version: '1.0.0',
}));

const getTransformCacheKeyFn = require('../getTransformCacheKeyFn');

describe('getTransformCacheKeyFn', () => {
  it('Should return always the same key for the same params', async () => {
    expect(
      getTransformCacheKeyFn({
        asyncRequireModulePath: 'beep',
        cacheVersion: '1.0',
        dynamicDepsInPackages: 'arbitrary',
        projectRoots: [__dirname],
        transformModulePath: require.resolve('metro/src/defaultTransform.js'),
      })(),
    ).toMatchSnapshot();
  });

  it('Should return a different key when the params change', async () => {
    const baseParams = {
      asyncRequireModulePath: 'beep',
      cacheVersion: '1.0',
      dynamicDepsInPackages: 'arbitrary',
      projectRoots: [__dirname],
      transformModulePath: require.resolve('metro/src/defaultTransform.js'),
    };

    const changedParams = [
      {
        ...baseParams,
        cacheVersion: '1.1',
      },
      {
        ...baseParams,
        projectRoots: ['/foo'],
      },
      {
        ...baseParams,
        transformModulePath: require.resolve(
          'metro/src/reactNativeTransformer',
        ),
      },
    ];

    const baseCacheKey = getTransformCacheKeyFn(baseParams)();

    changedParams.forEach(params => {
      expect(getTransformCacheKeyFn(params)()).not.toEqual(baseCacheKey);
    });
  });
});
