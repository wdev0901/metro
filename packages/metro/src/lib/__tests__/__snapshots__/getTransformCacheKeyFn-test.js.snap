// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTransformCacheKeyFn Should return always the same key for the same params 1`] = `"1c70ddd6a187bff45cc2404e694d377a51d1a28b"`;
